### ISSUE TITLE
Don't support optional positional parameters?

### ISSUE THREAD

**Issue by taosimple** [ethanblake4/dart_eval#99]
```
num ADD([num a = 0, num b = 0]){
  return (a + b);
}
```
Eval above codes throws exception:
```
The following CompileError was thrown while handling a gesture:
Unknown formal type DefaultFormalParameterImpl at unknown (file package:default/main.dart)

When the exception was thrown, this was the stack: 
#0      Compiler.compileSources (package:dart_eval/src/eval/compiler/compiler.dart:394:7)
#1      Compiler.compile (package:dart_eval/src/eval/compiler/compiler.dart:142:12)
#2      eval (package:dart_eval/src/eval/eval.dart:35:28)

```

**Comment by taosimple** [comment]
Fix: 
In `lib/src/eval/compiler/expression/function.dart:102` change:
```
final positionalArgTypes = positional
      .cast<SimpleFormalParameter>()
      .map((a) => a.type == null ? EvalTypes.dynamicType : TypeRef.fromAnnotation(ctx, ctx.library, a.type!))
      .map((t) => t.toRuntimeType(ctx))
      .map((rt) => rt.toJson())
      .toList();
  ```
  to:
  ```
    final positionalArgTypes = positional
      .map((a) {
        if(a is SimpleFormalParameter){
          return a.type == null ? EvalTypes.dynamicType : TypeRef.fromAnnotation(ctx, ctx.library, a.type!);
        }else{
          return EvalTypes.dynamicType;
        }
  })
      .map((t) => t.toRuntimeType(ctx))
      .map((rt) => rt.toJson())
      .toList()
  ```
  But lost parameters type

**Comment by ethanblake4** [comment]
Fixed in v0.6.2
